src/core/meshcop/commissioner.cpp

```diff
@@ -400,9 +410,10 @@ void Commissioner::HandleUdpTransmit(void)
     mTransmitMessage->Write(mTransmitMessage->GetOffset(), sizeof(tlv), &tlv);
 
     memset(&messageInfo, 0, sizeof(messageInfo));
-    messageInfo.GetPeerAddr() = mJoinerRouterAddress;
+    messageInfo.GetPeerAddr() = *mNetif.GetMle().GetMeshLocal16();
+    messageInfo.GetPeerAddr().mFields.m16[7] = HostSwap16(mJoinerRloc);
```

Comment:
This construct is less beautiful than the rest.  Is there a way to have a nicely named helper function such as .SetRloc16() do this and handle the magic access to offset [7]?